VDA-StorageNode-Registry/contracts/StorageNodeRegistry.sol

```diff
@@ -0,0 +1,1098 @@
+     */
+    struct DatacenterInfo {
+        bool isActive;
+        uint nodeCount;
+    }
+
+    /**
+     * @notice Additional information related to staking slots
+     * @dev Used internally inside the contract
+     * @param isStakingRequired true if staking required, otherwise false
+     * @param STAKE_PER_SLOT The number of tokens required to stake for one storage slot.
+     * @param MIN_SLOTS The minimum value of `STAKE_PER_SLOT`
+     * @param MAX_SLOTS The maximum value of `STAKE_PER_SLOT`
+     * @param NODE_ISSUE_FEE The number of VDA tokens that must be deposited when recording an issue against a storage node. `default=5`
+     * @param SAME_NODE_LOG_DURATION Time after which log available for same node
+     */
+    struct SlotInfo {
+        bool isStakingRequired;
+        uint STAKE_PER_SLOT;
+        uint MIN_SLOTS;
+        uint MAX_SLOTS;
+
+        uint NODE_ISSUE_FEE;
+        uint SAME_NODE_LOG_DURATION;
+
+        uint totalIssueFee;
+    }
+
+    /**
+     * @notice Represent a each log information
+     * @dev Every DID keeps 4 DIDLogInformations to restrict 4 logs in 24 hours
+     * @param nodeDID Node DID from `nodeLogIssue()` function
+     * @param reasonCode Reason code from `nodeLogIssue()` function
+     * @param time Issue logged time
+     */
+    struct IssueInformation {
+        address nodeDID;
+        uint reasonCode;
+        uint time;
+    }
+
+    /**
+     * @notice Represent the log list for a DID
+     * @dev It keeps last 4 logs recorded by `logNodeIssue()` function
+     * @param _issueList Issue information list
+     * @param index The earliest issue index in the list
+     */
+    struct DIDLogInformation {
+        IssueInformation[] _issueList;
+        uint index;
+    }
+
+    error InvalidDatacenterName();
+    error InvalidCountryCode();
+    error InvalidRegionCode();
+    error InvalidLatitude();
+    error InvalidLongitude();
+    error InvalidDatacenterId(uint id);
+    error HasDependingNodes();
+    error InvalidEndpointUri();
+    error InvalidDIDAddress();
+    error InvalidUnregisterTime();
+    error InvalidTokenAddress();
+    error InvalidSlotCount();
+    error InvalidValue();
+    error NoExcessTokenAmount();
+    error TimeNotElapsed(); // 4 logs in 24 hour
+    error InvalidSameNodeTime(); 
+    error InvalidAmount();
+    error InvalidReasonCode();
+
+    // constructor() {
+    //     _disableInitializers();
+    // }
+
+    /**
+     * @dev initializer of deployment
+     */
+    function initialize(address tokenAddress) initializer public {
+        __VDAVerificationContract_init();
+
+        _slotInfo.STAKE_PER_SLOT = 3 * (10 ** ERC20Upgradeable(tokenAddress).decimals());
+        _slotInfo.isStakingRequired = false;
+        _slotInfo.MIN_SLOTS = 20000;
+        _slotInfo.MAX_SLOTS = 20000;
+
+        _slotInfo.NODE_ISSUE_FEE = 5 * (10 ** ERC20Upgradeable(tokenAddress).decimals());
+        _slotInfo.SAME_NODE_LOG_DURATION = 1 hours;
```

Comment:
`Accounts can not log more than 4 issues against any node for a given 24 hour period.`

This requirement seems to be missing?

Reply:
Pls check the line#1017 in the `StorageNodeRegistry.sol` file.
I didn't define a constant for this `4` in the contract.

Reply:
Updated.
Added `LOG_LIMIT_PER_DAY` variable in the `SlotInfo` struct.